tests/test_eks/test_eks.py

```diff
@@ -0,0 +1,770 @@
+
+
+def test_create_nodegroup_throws_exception_when_nodegroup_already_exists(
+    NodegroupBuilder,
+):
+    client, generated_test_data = NodegroupBuilder(BatchCountSize.SMALL)
+    expected_exception = ResourceInUseException
+    expected_msg = NODEGROUP_EXISTS_MSG.format(
+        exception_name=expected_exception.TYPE,
+        method=MethodNames.CREATE_NODEGROUP,
+        cluster_name=generated_test_data.cluster_name,
+        nodegroup_name=generated_test_data.existing_nodegroup_name,
+    )
+
+    client.create_nodegroup.when.called_with(
+        clusterName=generated_test_data.cluster_name,
+        nodegroupName=generated_test_data.existing_nodegroup_name,
+        **dict(NodegroupInputs.REQUIRED)
+    ).should.throw(expected_exception, expected_msg)
+    len(
+        client.list_nodegroups(clusterName=generated_test_data.cluster_name)[
+            ResponseAttribute.NODEGROUPS
+        ]
+    ).should.equal(BatchCountSize.SMALL)
+
+
+def test_create_nodegroup_throws_exception_when_cluster_not_active(
+    NodegroupBuilder, monkeypatch
+):
+    client, generated_test_data = NodegroupBuilder(BatchCountSize.SMALL)
+    expected_exception = InvalidRequestException
+    expected_msg = CLUSTER_NOT_READY_MSG.format(
+        exception_name=expected_exception.TYPE,
+        method=MethodNames.CREATE_NODEGROUP,
+        cluster_name=generated_test_data.cluster_name,
+    )
+
+    with mock.patch("moto.eks.models.Cluster.isActive", return_value=False):
+        client.create_nodegroup.when.called_with(
+            clusterName=generated_test_data.cluster_name,
+            nodegroupName=random_string(),
+            **dict(NodegroupInputs.REQUIRED)
+        ).should.throw(expected_exception, expected_msg)
+
+    len(
+        client.list_nodegroups(clusterName=generated_test_data.cluster_name)[
+            ResponseAttribute.NODEGROUPS
+        ]
+    ).should.equal(BatchCountSize.SMALL)
+
+
+def test_create_nodegroup_generates_valid_nodegroup_arn(NodegroupBuilder):
+    _, generated_test_data = NodegroupBuilder()
+    expected_arn_values = [
+        PARTITIONS,
+        REGION,
+        ACCOUNT_ID,
+        generated_test_data.cluster_name,
+        generated_test_data.nodegroup_names,
+        None,
+    ]
+
+    all_arn_values_should_be_valid(
+        expected_arn_values=expected_arn_values,
+        pattern=RegExTemplates.NODEGROUP_ARN,
+        arn_under_test=generated_test_data.nodegroup_describe_output[
+            NodegroupAttributes.ARN
+        ],
+    )
+
+
+@pytest.mark.freeze_time
+def test_create_nodegroup_generates_valid_nodegroup_created_timestamp(NodegroupBuilder):
+    _, generated_test_data = NodegroupBuilder()
+    current_time = iso_8601_datetime_without_milliseconds(datetime.now())
+
+    result_time = iso_8601_datetime_without_milliseconds(
+        generated_test_data.nodegroup_describe_output[NodegroupAttributes.CREATED_AT]
+    )
+
+    result_time.should.equal(current_time)
```

Comment:
Note that freezing time will not work in ServerMode, as that will start Moto in a completely separated Docker container.
You can use settings.TEST_SERVER_MODE to guard the assertions, and maybe just check that the attribute exists/has the correct format in server mode. (Example: https://github.com/spulec/moto/blob/master/tests/test_sts/test_sts.py#L85